parse/src/test/java/com/parse/ParseObjectTest.java

```diff
@@ -137,6 +138,30 @@ public void testFromJSONPayloadWithoutClassname() throws JSONException {
         assertNull(parseObject);
     }
 
+    @Test
+    public void testFromJsonWithLdsStackOverflow() throws JSONException {
+        ParseObject localObj = ParseObject.createWithoutData("GameScore", "TT1ZskATqS");
+        OfflineStore lds = mock(OfflineStore.class);
+        Parse.setLocalDatastore(lds);
+
+        when(lds.getObject(eq("GameScore"), eq("TT1ZskATqS"))).thenReturn(localObj);
+
+        JSONObject json = new JSONObject("{" +
+                "\"className\":\"GameScore\"," +
+                "\"createdAt\":\"2015-06-22T21:23:41.733Z\"," +
+                "\"objectId\":\"TT1ZskATqS\"," +
+                "\"updatedAt\":\"2015-06-22T22:06:18.104Z\"" +
+                "}");
+        ParseObject obj;
+        for (int i = 0; i < 50000; i++) {
+            obj = ParseObject.fromJSON(json, "GameScore", ParseDecoder.get(), Collections.<String>emptySet());
+        }
+    }
+
+    //endregion
+
+    //region testGetter
```

Comment:
Can you change testGetter to testFromJson so they match?

Reply:
I moved `testFromJSONPayloadWithoutClassname` into the new `testFromJson` region.  This is just the other part of that move.  The `testFromJson` region ended at line 161 with `//endregion`

Reply:
Ahhh gotcha, thanks